distsql: memory panic when sorting with a limit

Running `CockroachDB CCL v2.0-alpha.20180212-398-g295c04f` results in a leftover bytes panic:
```
panic: panic while executing 1 statements: SELECT * FROM _._ ORDER BY _ LIMIT _; caused by flow: unexpected 647135 leftover bytes
```
This was running `SELECT * FROM tpch.lineitem ORDER BY l_extendedprice LIMIT 1000`. This is most probably related to @arjunravinarayan's recent sorter change.

--- a/pkg/sql/distsqlrun/base.go
+++ b/pkg/sql/distsqlrun/base.go
@@ -150,20 +150,18 @@ func Run(ctx context.Context, src RowSource, dst RowReceiver) {
 			case NeedMoreRows:
 				continue
 			case DrainRequested:
-				row = nil
+				DrainAndForwardMetadata(ctx, src, dst)
+				dst.ProducerDone()
+				return
 			case ConsumerClosed:
 				src.ConsumerClosed()
 				dst.ProducerDone()
 				return
 			}
 		}
-		if row == nil {
-			if meta != nil {
-				DrainAndForwardMetadata(ctx, src, dst)
-			}
-			dst.ProducerDone()
-			return
-		}
+		// row == nil && meta == nil: the source has been fully drained.
+		dst.ProducerDone()
+		return
 	}
 }
 

